toolchain/sem_ir/ids.h

```diff
@@ -86,6 +86,20 @@ constexpr InstId InstId::Invalid = InstId(InvalidIndex);
       InstId::ForBuiltin(BuiltinInstKind::Name);
 #include "toolchain/sem_ir/builtin_inst_kind.def"
 
+// An ID of an instruction that is referenced absolutely within a typed
+// instruction. This means that the instruction always represents the ID of a
+// global entity that is independent of the current context, and operations like
+// substitution into the enclosing instruction should not substitute into fields
+// with this type.
+class AbsoluteInstId : public InstId {
+ public:
+  // Implicitly converts from the base class.
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  constexpr AbsoluteInstId(InstId inst_id) : InstId(inst_id) {}
```

Comment:
The implicit conversion makes me a little nervous, especially since the conversion in the other direction is implicit as well. What would we lose if it was explicit?

Reply:
It'd just introduce noise into all the places that construct typed insts, explicitly performing a conversion they don't care about.